python/lsst/drp/tasks/metadetection_shear.py

```diff
@@ -559,7 +564,8 @@ def runQuantum(
             config=self.config.ref_loader,
             log=self.log,
         )
-        ref_cat = ref_loader.loadRegion(qc.quantum.dataId.region)
+        # What should decide the filterName?
+        ref_cat = ref_loader.loadRegion(qc.quantum.dataId.region, filterName="r")  # THIS IS A HACK.
```

Comment:
I know it says it's a hack but can't you get the band from the dataId?

Reply:
This is a multi-band task, so `dataId` doesn't have a band.

Reply:
You do know all possible bands in play though because you calculate them in the next line. I have no idea how you guess which of the known bands is the one that you choose for the refcat loader though.

Reply:
Exactly. I don't know either. And it might not matter at all for this purpose if the downstream code (which is `metadetect` here) does not rely on a reference band for fluxes (which I don't think it does)

Reply:
correct, it does not